### PR TITLE
HPCC-19235 Remove timestamps from regression suite output

### PR DESCRIPTION
**Pull request by ghalliday**
Remove times from exceptions, and some other details that change between
runs and versions.

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by ghalliday** [comment]
The aim of this PR is to allow binary comparison of regression suite results.  The parameter that has been renamed is only used when running the regression suite.  Should have no effect on normal system use to suitable for a point release.

@shamser please review.

File: common/workunit/workunit.cpp [hunk 1]
```diff
@@ -11568,28 +11568,45 @@ extern WORKUNIT_API StringBuffer &exportWorkUnitToXML(const IConstWorkUnit *wu,
...
+        if (regressionTest)
         {
-            StringBuffer statsFilename;
-            statsFilename.append(filename).append(".stats");
+            //This removes any items from the xml that will vary from run to run, so they can be binary compared from run to run
+            //The following attributes change with the build.  Simpler to remove rather than needing to updated each build
+            p->removeProp("@buildVersion");
+            p->removeProp("@eclVersion");
+            p->removeProp("@hash");
+
```
Comment (shamser) [inline common/workunit/workunit.cpp]: I don't see it removing any 'ts' attributes or statistic  types such as WhenStarted and TimeElapsed.  For a binary compare should these be removed to.  I'm referring the stats output in the following format:
<Statistics>
 <Statistic c="eclcc" count="1" creator="eclcc" kind="WhenStarted" s="compile" scope="compile" **ts="1541063347021154"** **unit="ts"** **value="1541063347021131"**/>
 <Statistic c="eclcc" count="1" creator="eclcc" kind="WhenStarted" s="compile" scope="compile:parse" **ts="1541063347021212" **unit="ts"** **value="1541063347021202"**/>
 <Statistic c="eclcc" count="1" creator="eclcc" kind="TimeElapsed" s="compile" scope="compile:parse" **ts="1541063353737173" **unit="ns"** **value="6719330841"**/>
 <Statistic c="eclcc" count="1" creator="eclcc" kind="NumAttribsSimplified" s="compile" scope="compile:cache" **ts="1541063353737203"** unit="cnt" value="0"/>
 <Statistic c="eclcc" count="1" creator="eclcc" kind="NumAttribsProcessed" s="compile" scope="compile:cache" **ts="1541063353737233"** unit="cnt" value="250"/>
 <Statistic c="eclcc" count="1" creator="eclcc" kind="NumAttribsFromCache" s="compile" scope="compile:cache" **ts="1541063353737264"** unit="cnt" value="0"/>
 <Statistic c="eclcc" count="1" creator="eclcc" kind="NumAttribsSimplifiedTooComplex" s="compile" scope="compile:cache" **ts="1541063353737301"** unit="cnt" value="0"/>
</Statistics>

File: ecl/hqlcpp/hqlcerrors.hpp [hunk 2]
```diff
@@ -170,7 +170,7 @@
 #define HQLERR_CouldNotFindDataset              4158
 #define HQLERR_CouldNotAnyDatasetX              4159
 #define HQLERR_NestedThorNodes                  4160
-#define HQLERR_MissingTransformAssignXX         4161
+#define HQLERR_MissingTransformAssignX          4161
```
Comment (shamser) [inline ecl/hqlcpp/hqlcerrors.hpp]: Is this and the changes below related to this issue?

**Review by shamser — COMMENTED** [review]
@ghalliday

File: common/workunit/workunit.cpp (hunk 1 above)
Comment (ghalliday) [reply common/workunit/workunit.cpp]: All of the statistics are removed - see the next lines, and extracted to a separate file.

File: ecl/hqlcpp/hqlcerrors.hpp (hunk 2 above)
Comment (ghalliday) [reply ecl/hqlcpp/hqlcerrors.hpp]: Yes, it was tracing a pointer value in one of the error messages, so I remove it.  XX indicated there were two arguments, so removed one from the constant name.

**Comment by ghalliday** [comment]
@shamser see replies

**Review by shamser — APPROVED** [review]
@richardkchapman Code reviewed. Looks good.

**Pull request merged**